ocaml/typing/jkind_types.mli

```diff
@@ -152,8 +152,21 @@ module Type : sig
   type 'type_expr annotation = 'type_expr Const.t * Jane_syntax.Jkind.annotation
 end
 
-type 'type_expr t = 'type_expr Type.t
+module Arrow : sig
+  type 'a t =
+    { args : 'a list;
+      result : 'a
+    }
+end
+
+type 'type_expr t =
+  | Type of 'type_expr Type.t
+  | Arrow of 'type_expr t Arrow.t
```

Comment:
I'd probably just inline the definition of `Arrow.t` here. I don't think we gain much from splitting it out.

Reply:
I actually sort of like `Arrow.t`, as it's reused semi-often and it's nicer to have a type definition on-hand. It could be argued that `'type_expr t list * 'type_expr t` is already clear in being an n-ary type constructor signature, though. But if it is to be a record type, I'd prefer to keep it this way.

Reply:
Once it is removed from `out_jkind_*`s, `Arrow.t` is only used for `Jkind_types.t`, `Jkind_types.Const.t`, `Jkind.Desc.t` (`get`) and is passed around in a couple places. So I could remove it. Should it remain a record type or become a product type?

Reply:
This conversation is marked as resolved, but it looks like the `Arrow.t` type is still here.

Reply:
Sorry - I assumed that since you said 'it's nice to have names for things' you agreed to have `Arrow.t` exist? Or were you just referring to field names?